activerecord/lib/active_record/reflection.rb

```diff
@@ -958,10 +952,6 @@ def collect_join_reflections(seed)
 
       protected
 
-        def actual_source_reflection # FIXME: this is a horrible name
-          source_reflection.send(:actual_source_reflection)
-        end
-
```

Comment:
This no longer recurses, then, huh?

https://github.com/maclover7/rails/commit/b8aebb0261470af3df4e55d125b1ce8e8bf1cb32